### PR TITLE
Update printf(3) based on latest OpenBSD structure and format

### PR DESCRIPTION
**Pull request by chrisdavidson**
Bug #247900
   https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=247900

printf.3
   Update order of parameters alphabetically
   Add parameter syntax to each parameter for clarity
   Untangle multiple option entries into individual entries
   Update sentence structure throughout
   Provide clarity on symbol based parameters

### CODE REVIEW

**Comment by chrisdavidson** [comment]
I noticed another branch updates got put into this pull request. I am working on REMOVING the files associated with 4743e9f.

**Comment by dag-erling** [comment]
The entries for the `B` and `b` conversion specifiers are incorrect, cf. https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2630.pdf

**Comment by chrisdavidson** [comment]
> The entries for the `B` and `b` conversion specifiers are incorrect, cf. https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2630.pdf

The referenced document highlights unsigned int to unsigned binary conversion. It looks like the original manual page did not have this reference, i will update accordingly.

**Comment by dag-erling** [comment]
> The referenced document highlights unsigned int to unsigned binary conversion. It looks like the original manual page did not have this reference, i will update accordingly.

You are mistaken, cf. 76edfabb.

**Comment by chrisdavidson** [comment]
> > The referenced document highlights unsigned int to unsigned binary conversion. It looks like the original manual page did not have this reference, i will update accordingly.
> 
> You are mistaken, cf. [76edfab](https://github.com/freebsd/freebsd-src/commit/76edfabbecdec686a570b8e009d5ea4112f943e0).

I looked through that update and I must be missing something. 

The options b and B respectively are being described and when used with the x and X conversion parameters respectively they work as described. 

The updates conducted in this pull request are moving the elements to be more concise and easier to read. The changes that were conducted in August 2023, as referenced by the commit id above, are included in the baseline that the updates were based on, dated 05SEP2023.

**Comment by dag-erling** [comment]
I am the author of both the commit in question and the code it documents. I assure you that I know what it is about. There is no such thing as “B and b options used with the X and x conversion parameters”. `X` and `x` are conversion _specifiers_, and so are `B` and `b`.

**Comment by chrisdavidson** [comment]
okay, i will re-review and make appropriate changes.

beyond that, is there anything else that you thought was misaligned?

**Comment by bsdimp** [comment]
generally this is great, let's get the issues resolved so we can commit.

**Comment by chrisdavidson** [comment]
let me bring this back up.. the only issue i am aware of is the options mentioned? are there any other ones i should review?

**Comment by dag-erling** [comment]
Yes, you have 48 lines ending in  a spurious `g`, you should revert to 4ecdab4 if you still have it.

**Comment by dag-erling** [comment]
Now you have 48 lines ending in trailing whitespace.

**Comment by dag-erling** [comment]
Chris, please, slow down. You just pushed a bunch of conflict markers.

File: lib/libc/stdio/printf.3 [hunk 1]
```diff
@@ -105,12 +110,14 @@ and
...
 .Fa str ;
-and
 .Fn asprintf
 and
 .Fn vasprintf
 dynamically allocate a new string with
-.Xr malloc 3 .
+.Xr malloc 3
+that is stored in
+>>>>>>> 5affdd06c021 (Update order of parameters alphabetically)
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:119]: conflict marker

File: lib/libc/stdio/printf.3 [hunk 2]
```diff
@@ -606,28 +644,300 @@ and the (potentially multi-byte) sequence representing the
...
+.Xr strerror 3 .
+No argument is taken.
+.It Cm n
+The number of characters written so far is stored into the
+integer indicated by the
+.Vt "int *"
+(or variant) pointer argument.
+No argument is converted.
+.It Cm O
+A deprecate alias forg
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:913]:
```suggestion
A deprecated alias for
```

File: lib/libc/stdio/printf.3 [hunk 3]
```diff
@@ -606,28 +644,300 @@ and the (potentially multi-byte) sequence representing the
...
+.Ar lo .
+.It Cm p
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:915]: Missing an entire section for `.It Cm o` here.

File: lib/libc/stdio/printf.3 [hunk 4]
```diff
@@ -606,28 +644,300 @@ and the (potentially multi-byte) sequence representing the
...
+The
+.Vt "void *"
+pointer argument is printed ing
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:918]: trailing `g` again

File: lib/libc/stdio/printf.3 [hunk 5]
```diff
@@ -583,6 +605,22 @@ Zeroes are always represented with a mantissa of 0 (preceded by a
...
 if appropriate) and an exponent of
 .Li +0 .
+.Pp
+.It Cm b
+The
+.Vt unsigned int
+(or appropriate variant) argument is converted to unsigned binary.
+This is used for
+.Cm x
+conversions.
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:615]: no

File: lib/libc/stdio/printf.3 [hunk 6]
```diff
@@ -583,6 +605,22 @@ Zeroes are always represented with a mantissa of 0 (preceded by a
...
+.Pp
+.It Cm B
+The
+.Vt unsigned int
+(or apporiate variant) argument is converted to unsigned binary.
+This is used for
+.Cm X
+conversions.
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:623]: no

File: lib/libc/stdio/printf.3 [hunk 7]
```diff
@@ -583,6 +605,22 @@ Zeroes are always represented with a mantissa of 0 (preceded by a
 .Ql -
 if appropriate) and an exponent of
 .Li +0 .
+.Pp
+.It Cm b
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:609]: What about `argno`, `width`, `size` etc?

File: lib/libc/stdio/printf.3 [hunk 8]
```diff
@@ -583,6 +605,22 @@ Zeroes are always represented with a mantissa of 0 (preceded by a
...
+The
+.Vt unsigned int
+(or appropriate variant) argument is converted to unsigned binary.
+This is used for
+.Cm x
+conversions.
+.Pp
+.It Cm B
```
Comment (dag-erling) [inline lib/libc/stdio/printf.3:617]: What about `argno`, `width`, `size` etc?

**Comment by chrisdavidson** [comment]
bah, sorry about this, my local git is pushing to here and NOT to my local branch.

**Review by dag-erling — CHANGES_REQUESTED** [review]

**Comment by chrisdavidson** [comment]
i'm working through how to get this all merged into one commit. I will update the comments section upon completion of this consolidation efforts.

**Comment by chrisdavidson** [comment]
I closed this pull request as the more i worked on it, the worse it got. I am going to review and generate a new pull request. Thank you for the input everyone.

**Comment by dag-erling** [comment]
I'm sorry you feel that way, it seemed to be moving in the right direction.

**Comment by chrisdavidson** [comment]
It is not the content i was struggling with, it was my poor attempt to consolidate all the changes into one commit and not cause the massive conflicts that were occurring, unintentionally. I took the necessary steps to diff the whole commit and made a new branch, the results of this updates are available here: https://github.com/freebsd/freebsd-src/pull/1200

**Pull request closed without merging**